lib/engine/game/g_steam_over_holland/game.rb

```diff
@@ -422,7 +422,7 @@ def redeemable_shares(entity)
         end
 
         def emergency_issuable_bundles(entity)
-          return [] if @round.issued_shares[entity] || entity.cash >= @depot.min_depot_price
+          return [] if @round.issued_shares[entity] || entity.cash >= @depot.min_price(ability: nil)
```

Comment:
The `min_price` method has a `corporation` parameter. Omitting this parameter is probably going to cause an exception when run in server mode (and doesn't make any sense, if you're not trying to find the cheapest train for a specific corporation, then `depot_min_price` is the right method to use).
https://github.com/tobymao/18xx/blob/38a729965ef394d5cd3ace16e45952672fa64aac/lib/engine/depot.rb#L54-L56

Reply:
Yep, 100% right. I meant to include that. Thanks!

Reply:
Does the train discount from the private still apply if you're buying a train during emergency money raising? I think you will need to change the call to `min_price` for this to work: `@depot.min_price(entity, abilities(entity, :train_discount)`.

Reply:
When I change it to this, I get the following error in the console: 

```
ArgumentError: expected kwargs
    at Opal.send2 (deps.js:1958:17)
    at Opal.send (deps.js:1946:17)
    at Function.$$exception (deps.js:6052:14)
    at Function.$$raise (deps.js:5653:31)
    at $raise (deps.js:146:16)
    at Opal.ensure_kwargs (deps.js:2942:7)
    at klass.$$min_price (main.js:6790:19)
    at klass.$$emergency_issuable_bundles (g_steam_over_holland.js:414:116)
    at klass.$$issuable_shares (main.js:30539:28)
    at klass.$$render (main.js:49656:66)
    at klass.$$h (deps.js:27934:28)
    at klass.$$render (main.js:56432:34)
    at klass.$$h (deps.js:27934:28)
    at klass.$$render_action (main.js:61784:25)
    at klass.$$render_game (main.js:61831:30)
    at klass.$$render (main.js:61460:74)
    at klass.$$h (deps.js:27934:28)
    at klass.$$render_game (main.js:38001:19)
    at klass.$$render_content (main.js:37969:275)
    at klass.$$render (main.js:37957:154)
    at klass.$Component_update$excl$2 (deps.js:27962:26)
    at deps.js:27951:109
```

Reply:
Yes, my mistake. It should be `@depot.min_price(entity, ability: abilities(entity, :train_discount)`.

The "expected kwargs" error means we tried passing a parameter when a named parameter was needed.

Reply:
hmmm something's wrong now. I did a print of @depot.min_price(entity,ability: abilities(entity,:train_discount)), and it returns 1, rather than the price of the cheapest depot train minus 10%.

I'm gonna have to take a look at this in a couple of days. I'm going to be out all weekend.

Reply:
taking a quick look, even my code before was returning 1. So I'm still not pulling from the right spot. Will investigate more in a couple days.

Reply:
1 will be the price for buying a train from another company.

Reply:
yep. I submitted a new update, but it's still causing issues. the battle continues.